### PR TITLE
Feature/65 registry reload

### LINKED ISSUE #65: UI deployment
After #19 #18

# Description
Add UI deployment with Terraform.
Might need to be refined after #18 and after #19

# Acceptance criteria
- [ ] UI can be deployed with other MVD components
- [ ] ~Steps to deploy and access the UI are documented~ (will be done in #19, since currently the build destroys the environment right away)
- [ ] UI is deployed in CD to test the terraform
- [ ] The UI can be used and can call the backend

# Tasks
- [ ] ~Build and push image to ACR in build~ (will be done in https://github.com/agera-edc/EDCDataDashboard/issues/1)
- [ ] Add UI deployment to Terraform
- [ ] ~Document steps to deploy the UI~
- [ ] Set up CORS
- [ ] Set backend URL

### LINKED ISSUE #94: UI deployment

### PR DESCRIPTION
**Pull request by algattik**
Now that https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1243 got merged upstream, EDC can dynamically reload the registry nodes.

- Changed custom registry extension to reload JSON registry files when the federated catalog refreshes (by default every 10 minutes).
- Removed ACI restart task

Closes agera-edc/MinimumViableDataspaceFork#94

### CODE REVIEW

File: extensions/refresh-catalog/src/main/java/org/eclipse/dataspaceconnector/mvd/JsonFileBasedNodeDirectory.java [hunk 1]
```diff
@@ -23,65 +23,63 @@
...
+                    .collect(Collectors.toList());
         } catch (IOException e) {
             throw new EdcException(e);
         }
     }
 
-    private void insertNode(FederatedCacheNode node) {
-        monitor.info("Adding catalog node " + node.getName());
-        nodeDirectory.insert(node);
+    @Override
```
Comment (chrislomonico) [inline extensions/refresh-catalog/src/main/java/org/eclipse/dataspaceconnector/mvd/JsonFileBasedNodeDirectory.java:81]: Is there any plan to support this in the future or letting it throw the exception as there is no need to insert nodes for the node directory?

File: extensions/refresh-catalog/src/test/java/org/eclipse/dataspaceconnector/mvd/JsonFileBasedNodeDirectoryTest.java [hunk 2]
```diff
@@ -14,35 +14,31 @@
...
+import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.ArgumentMatchers.argThat;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
 
-class RefreshCatalogServiceTest {
+class JsonFileBasedNodeDirectoryTest {
```
Comment (chrislomonico) [inline extensions/refresh-catalog/src/test/java/org/eclipse/dataspaceconnector/mvd/JsonFileBasedNodeDirectoryTest.java:28]: nit: Is it worth adding a test to ensure that insert throws an exception?

**Review by chrislomonico — COMMENTED** [review]
General comments, if considered resolved, will approve

**Review by chrislomonico — APPROVED** [review]
covered comments with @algattik, marking as resolved

**Review by cpeeyush — APPROVED** [review]

**Pull request merged**